### PR TITLE
[ENH] Added node API vocabulary endpoints and logic

### LINKED ISSUE #12: Add node API vocabulary endpoints and logic to send requests to specified APIs
As the query tool, talking to the federation-API should be the same as talking to a node-API. The only difference is that from a federation-API I get a potentially longer list of answers from. 

Therefore
- endpoints and query parameters should be called the same as in https://github.com/neurobagel/api
- response shemas on these endpoints should be the same

### PR DESCRIPTION
**Pull request by rmanaem**
Closes #12

Changes proposed in this pull request:

- Modified the format of the nodes in the `NB_NDOES` env var to not include the endpoint  
- Implemented `get_terms_labels` as a generic function to handle requests from both `/{data_element_URI}/vocab` and `/{data_element_URI}/vocab` endpoints to avoid code duplication
- The response schemas for now are simple concatenations of responses from each node and therefore includes unnecessary repetition
  - They are kept in this format in case we'd like to have `node_url` as part of the response schema (not implemented here as it was deemed important enough to have its own PR)

## Checklist

- [x] PR has an interpretable title with a prefix (`[ENH]`, `[FIX]`, `[REF]`, `[TST]`, `[CI]`, `[MNT]`, `[INF]`, `[MODEL]`, `[DOC]`) _(see https://neurobagel.org/contributing/pull_requests for more info)_
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [x] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions.

### CODE REVIEW

File: app/api/crud.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ async def get(
 
     for node_url in util.parse_nodes_as_list(util.NEUROBAGEL_NODES):
         response = httpx.get(
-            url=node_url,
+            url=node_url+"query/",
```
Comment (surchs) [inline app/api/crud.py:68]: :cherries: but: 
```suggestion
            url=node_url + "query/",
```

File: app/api/crud.py [hunk 2]
```diff
@@ -82,3 +82,29 @@ async def get(
...
     return cross_node_results
+
+async def get_terms_labels(data_element_URI: str, labels: bool):
+    """Makes GET requests to one or more Neurobagel node APIs using httpx where the parameter is a Neurobagel variable."""
+
+    cross_node_results = []
+
+    for node_url in util.parse_nodes_as_list(util.NEUROBAGEL_NODES):
+        if labels:
+            url = node_url + "attributes/"+data_element_URI+"/vocab"
```
Comment (surchs) [inline app/api/crud.py:93]:
```suggestion
            url = node_url + "attributes/" + data_element_URI + "/vocab"
```
can you check that you have flake8 setup and running in your IDE?

File: app/api/crud.py [hunk 3]
```diff
@@ -82,3 +82,29 @@ async def get(
...
+        else:
+            url = node_url + "attributes/"+data_element_URI
```
Comment (surchs) [inline app/api/crud.py:95]:
```suggestion
            url = node_url + "attributes/" + data_element_URI
```

File: app/api/crud.py [hunk 4]
```diff
@@ -82,3 +82,29 @@ async def get(
...
+            follow_redirects=True,
+        )
+
+        if not response.is_success:
+            raise HTTPException(
+                status_code=response.status_code,
+                detail=f"{response.reason_phrase}: {response.text}",
+            )
+        
+        cross_node_results.append(response.json())
```
Comment (surchs) [inline app/api/crud.py:108]: This part here is pretty duplicate-y compared with the `get` method above. How about we factor that part out into its a generic `get` method and have the URL making business be taken care of by more aptly named functions?

File: app/api/utility.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 #  Neurobagel nodes
 NEUROBAGEL_NODES = os.environ.get(
-    "NB_NODES", "https://api.neurobagel.org/query/"
+    "NB_NODES", "https://api.neurobagel.org/"
```
Comment (surchs) [inline app/api/utility.py:7]: Do we need to update some docs somewhere as well to make sure users don't accidentally add the `/query` back in their `.env` files?

**Review by surchs — APPROVED** [review]
Thanks @rmanaem! I have one question about the `crud` module, take a look and see what you think. If you don't think we should address this now, could you make an issue so we can address it later?

With that good to go 

:cook:

File: app/api/utility.py (hunk 5 above)
Comment (rmanaem) [reply app/api/utility.py:7]: Yea good catch, I'll update the README.md.

File: README.md [hunk 6]
```diff
@@ -7,7 +7,7 @@ The URLs should be stored as a **space-separated** string.
 
 e.g.,
 ```bash
-NB_NODES="https://myfirstnode.org/query/ https://mysecondnode.org/query/"
+NB_NODES="https://myfirstnode.org/ https://mysecondnode.org/"
```
Comment (surchs) [inline README.md:10]: oh, good catch! I missed that

File: app/main.py [hunk 7]
```diff
@@ -2,10 +2,10 @@
 
 import uvicorn
 from fastapi import FastAPI
-from fastapi.responses import ORJSONResponse
 from fastapi.middleware.cors import CORSMiddleware
+from fastapi.responses import ORJSONResponse
 
-from .api.routers import query, attributes
+from .api.routers import attributes, query
```
Comment (surchs) [inline app/main.py:8]: is this automatic? if yes: great. if no: can it be automatic?

File: app/api/crud.py [hunk 8]
```diff
@@ -64,47 +61,40 @@ async def get(
...
-                status_code=response.status_code,
-                detail=f"{response.reason_phrase}: {response.text}",
-            )
-        
-        cross_node_results.append(response.json())
+        cross_node_results.append(response)
 
-    return cross_node_results
\ No newline at end of file
+    unique_terms = set(
```
Comment (surchs) [inline app/api/crud.py:95]: neat

File: app/api/crud.py [hunk 9]
```diff
@@ -64,47 +61,40 @@ async def get(
...
         )
 
-        if not response.is_success:
-            raise HTTPException(
-                status_code=response.status_code,
-                detail=f"{response.reason_phrase}: {response.text}",
-            )
-        
-        cross_node_results.append(response.json())
+        cross_node_results.append(response)
```
Comment (surchs) [inline app/api/crud.py:93]: this is we get a dictionary back, yes?

**Review by surchs — APPROVED** [review]
thanks a lot for all these changes @rmanaem! I left some minor comments, largely for my understanding.

Take a look, and then :cook:

File: app/main.py (hunk 7 above)
Comment (rmanaem) [reply app/main.py:8]: I believe so.

File: app/api/crud.py (hunk 9 above)
Comment (rmanaem) [reply app/api/crud.py:93]: Yes

**Comment by surchs** [comment]
:rocket: PR was released in [`v0.1.0`](https://github.com/neurobagel/federation-api/releases/tag/v0.1.0) :rocket:

**Pull request merged**